pkg/controller/datavolume-controller.go

```diff
@@ -438,6 +438,20 @@ func (r *DatavolumeReconciler) Reconcile(_ context.Context, req reconcile.Reques
 		return reconcile.Result{}, err
 	}
 
+	// When cloning a PVC, if the target's size is not specified,
+	// said value can be attainable from the source PVC
+	if selectedCloneStrategy != NoClone {
+		targetRequest, hasTargetRequest := pvcSpec.Resources.Requests[corev1.ResourceStorage]
+		if hasTargetRequest && targetRequest.IsZero() {
+			sourcePvc, _ := r.findSourcePvc(datavolume)
```

Comment:
currently you only use this value in the if so maybe assign it inside the if?

Reply:
It'll probably be also used inside the `else`, but I'm still not 100% sure.

Reply:
I wonder, wouldn't it be cleaner if this code is placed inside the `RenderPvcSpec`? The you would correctly render the size requirements of the target PVC spec at the beginning.

Reply:
The idea of this render function is, that given `Storage` or `PVC` fields or any other things from the DataVolume we can create a single consistent view of a Target PVC for CDI controllers to use.

Reply:
I agree, I'm not very happy including that code inside the reconcile, but we need to know the `selectedCloneStrategy` before assigning the storage size. I don't know if we have a better option.

Reply:
do you need other values than NoClone?

because `NoClone` is in practice equivalent to `spec.Source.PVC == nil`

Reply:
ok, I see, you need the `Smart/CsiClone`